### PR TITLE
Check on node startup if all ports are free and exit if they are not

### PR DESCRIPTION
**Pull request by sebtomba**
## Overview
Check on node startup if all ports are free and exit if they are not

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-732

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### CODE REVIEW

File: node/src/main/scala/coop/rchain/node/Main.scala [hunk 1]
```diff
@@ -121,13 +119,36 @@ object Main {
...
 
+  private def checkPorts(conf: Configuration): Task[Unit] = {
+    def isLocalPortAvailable(port: Int): Task[Boolean] =
+      Task
+        .delay(new java.net.ServerSocket(port).close())
+        .attempt
+        .map(_.isRight)
+        .ifM(
+          true.pure[Task],
+          Task.delay(System.err.println(s"Port $port is already in use!")).as(false)
```
Comment (itegulov) [inline node/src/main/scala/coop/rchain/node/Main.scala]: Why use `System.err.println` instead of `Log[F].error`?

Comment (sebtomba) [reply node/src/main/scala/coop/rchain/node/Main.scala]: @adaszko what do you prefer? On startup of the node, if something is wrong with the configuration we print a message to the `err` out and exit. Now we do the same with the ports. If one of the ports is not free we (would) print a message to `err` out and exit.

Comment (adaszko) [reply node/src/main/scala/coop/rchain/node/Main.scala]: Actually, it is the process exit code that is most important in this case.  It should be nonzero.

Assuming the exit code is nonzero, I'd tilt slightly towards `Log[F].error`, just for consistency (i.e. not to multiply output channels unnecessarily).

**Pull request merged**